### ISSUE TITLE
Device Hierarchy required 

### ISSUE THREAD

**Issue by stuartp44** [Sikerdebaard/hacs-ducobox-connector#2]
I have noted that this plugin treats the connectivity board as the main "device". Being that this platform is highly modular, we need to design the plugin to be module and discover devices as such.  I have drafted this image to try and show what we should try to achieve.

![image](https://github.com/user-attachments/assets/5f031bce-60c8-4190-ab77-b9c37e8a7d32)

The duco connectivity boards should be one of many devices that this system creates with entities.

**Comment by Sikerdebaard** [comment]
Yes, lets do this. My only issue is that I am somewhat time constrained at the moment and am not super familiar with implementing these types of things in Home Assistant.

Do you have a suggestion on how to move forward with this?

**Comment by stuartp44** [comment]
I have started work on getting this implemented on my fork but I am trying to bump processes for deprecated stuff as-well. I have done most of the behavior as described but need to do the dynamic discovery of the BOX sensors next, once I can actually fix most of the issues around loading

**Comment by stuartp44** [comment]
I have opened a brand image PR for the images for the domain. I am using the brand direct as the domain rather than using ducobox. https://github.com/home-assistant/brands/pull/6169

**Comment by Sikerdebaard** [comment]
@stuartp44 I am thinking of splitting the API in [Sikerdebaard/ducopy](https://github.com/Sikerdebaard/ducopy) into a RAW api (json) and a more device-oriented one where pydantic models are used to represent a device hierarchy as drawn in your diagram. I think this will help in interacting with the devices / nodes in this HACS addon and this will hopefully lead to a more clean codebase in this HACS addon.

Given that this is quite a refactor, I'd like to discuss this with you. Do you agree with this approach? Or should we go in a different direction?

**Comment by stuartp44** [comment]
It looks like a better cleaner solution, for me, go for it. We can just anchor the version of the library for now and just release the plugin as a BETA as is. I have yet to do some more work on my fork to get "set" commands to work and then hopefully automations are next.
